### DISCUSSION TITLE
How to set baseHref when using Rspack?

### DISCUSSION THREAD

**Discussion by iconag-bbasmer** [nrwl/nx#28076] [Questions]
I have lately updated to Nx version 19.7.4 and as I'm using module federation with React I had to go through quite a few things to correctly setup everything using the Rspack settings instead of Webpack. 

What I struggle with now, is in the project.json the baseHref setting under targets.build.configurations.production. 
With Webpack I could set there the path that got added as base href in the head if the index.html file. If I set this in my project now, the baseHref is being ignored. 

I have attached my project.json of the application to this message. Is this a bug or have I done anything wrong?

Thanks for any help.
[project.json](https://github.com/user-attachments/files/17115665/project.json)

**Comment by a-s-ml** [comment]
Hi

"targets": {
     "build": {
          "options": {
               "baseHref":

**Reply by iconag-bbasmer** [reply]
Hi.
Thanks for answering but no, unfortunately this doesn't work. I have added it in the project.json of one of my projects to the path you mentioned above but when I run "nx run <projectname>:build", it doesn't add it to the index.html file in the resulting build.

I even tried to add this to a completely new workspace and project but no luck. :-(

Also, previously I only added the baseHref to targets.build.configurations.production.baseHref so that it only puts it into production builds but this doesn't work anymore either.

My project.json of my example project looks like this:

```
{
      "name": "myhost",
      "$schema": "../../node_modules/nx/schemas/project-schema.json",
      "sourceRoot": "apps/myhost/src",
      "projectType": "application",
      "tags": [],
      "targets": {
        "build": {
          "executor": "@nx/rspack:rspack",
          "outputs": ["{options.outputPath}"],
          "defaultConfiguration": "production",
          "options": {
            "target": "web",
            "baseHref": "/path/to/prod",
            "outputPath": "dist/myhost",
            "main": "apps/myhost/src/main.ts",
            "tsConfig": "apps/myhost/tsconfig.app.json",
            "assets": ["apps/myhost/src/favicon.ico", "apps/myhost/src/assets"]
          },
          "configurations": {
        ...
```

Not sure what's going on here.

**Comment by scott-gmr** [comment]
This seems to be another Nx bug. I opened [this issue](https://github.com/nrwl/nx/issues/28455).
